CIP-0014/CIP-0014.md

```diff
@@ -0,0 +1,116 @@
+# CIP-0014 User-Facing Asset Id 
+
+# Preamble
+
+---
+CIP: 0014
+Title: User-Facing Asset Id
+Authors: Matthias Benkort <matthias.benkort@iohk.io>, Rodney Lorrimar <rodney.lorrimar@iohk.io>
+Comments-Summary: A common format for displaying a unique identifier for assets.
+Comments-URI: https://github.com/cardano-foundation/CIPs/wiki/Comments:CIP-0014
+Status: Draft
+Type: Standards
+Created: 2020-02-01
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This specification defines a unique user-facing asset id as a bech32-encoded blake2b-256 digest of the concatenation of the policy id and the asset name.
+
+# Motivation
+
+The Mary era of Cardano introduces the support for native assets. On the blockchain, native assets are uniquely identified by both their so-called policy id and asset name. Neither the 
+policy id nor the asset name are intended to be human-readable data. 
+
+On the one hand, the policy id is a hash digest of either a monetary script or a Plutus script. On the other hand, the asset name is an arbitrary bytestring of up to 32 bytes (which does not necessarily decode to a valid UTF-8 sequence). In addition, it is possible for an asset to have an empty asset name, or, for assets to have identical asset names under different policy ids. 
+
+Because assets are manipulated in several user-facing features, and also via hardware devices, it is useful to come up with a short(er) and human-readable identifier for assets that user can recognize and refer to when talking about assets.
+
+# Specification
+
+We define a unique identifier for assets in pseudo-code as:
+
+```
+assetId := encodeBech32
+  ( datapart = hash
+    ( algorithm = 'blake2b'
+    , digest-length = 32
+    , message = policyId | assetName
+    )
+  , humanReadablePart = 'token'
```

Comment:
is there a clear distinction between the term "asset" and "token" in Cardano terminology? I'm wondering because assets are being mentioned in the surrounding text whereas the bech32 prefix is "token".

Reply:
Well, it seems that both are used a bit interchangeably but the official documentation talks about "tokens". So, which is quite different from the ledger which refers to them more often as "assets". To me, "tokens" also suggests fungibility whereas "assets" tends to be more general.

Reply:
I see a clear distinction, which is expressed nicely in:

https://developers.cardano.org/en/development-environments/native-tokens/multi-asset-tokens-explainer/

> Assets
An asset is an object that represents value on the blockchain. These objects can be a variety of things, such as a digital asset like ada, a role, a credential, or a quantity of goods.

> Tokens
A token is a short term for “asset token”, which is the on-chain representation of an asset and its basic accounting unit. A token can represent one ada, one house, or the value of ten tonnes of coffee, for example.


Assets are transacted **_as_** tokens, even NFTs. token replaces coin. With this in mind I would vote for `asset` as the prefix

Reply:
Should we include the prefix in CIP5 as well?

Reply:
@mmahut eventually, yes!
There à debate however about the prefix too: token vs asset

Reply:
Did we reach a conclusion? Looks like Daedalus is using`token`.

Reply:
@KtorZ can you please confirm which one (`asset` or `token`) is final? I'm seeing both all over the place and I'm really not sure.

Reply:
@mmahut @1000101

I've settled on `asset` in the last version of the proposal. Daedalus will be using whatever `cardano-wallet` implements, which is `asset` so far. I hope we can maybe get this merge soon as a draft to "settle it".

Reply:
(sorry for the confusion, thought I had updated the remote but I apparently did push on the wrong repository last week :|)

Reply:
@KtorZ All good! Thanks a million!!! :)